test/s3path.jl

```diff
@@ -334,4 +334,13 @@ end
     end
 end
 
+@testset "JSON roundtripping" begin
+    json_path = S3Path("s3://$(bucket_name)/test_json"; config=aws)
+    my_dict = Dict("key" => "value", "key2" => 5.0)
+    s3_put(aws, bucket_name, "test_json", JSON3.write(my_dict), "application/json")
```

Comment:
here we use the `"application/json"` MIME type to trigger the heuristic parsing into a LittleDict that will hit a `MethodError` at the `Vector{UInt8}` constructor.

Reply:
I think it is worth adding this as comment here

Reply:
added!